src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -447,7 +504,11 @@ export class TerminalService implements ITerminalService {
 			}
 		} else if (this.terminalTabs.length === 0) {
 			// Local window, or remote terminal reconnection is disabled, just create a terminal
-			this.createTerminal();
+			if (this.configHelper.config.mainSpawn) {
+				this._reconnectToLocalTerminals();
```

Comment:
It's possible to have local and remote terminals at the same time, while less important I'd expect this to get restored as well.